### PR TITLE
feat: fav page references

### PR DESCRIPTION
**Pull request by pengx17**
https://github.com/toeverything/AFFiNE/assets/584378/a48b355f-906c-4942-8695-21f9d56415c9

### CODE REVIEW

File: packages/hooks/src/use-block-suite-page-references.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import type { Page, Workspace } from '@blocksuite/store';
+import { atom, useAtomValue } from 'jotai';
+import { atomFamily } from 'jotai/utils';
+
+import { useBlockSuiteWorkspacePage } from './use-block-suite-workspace-page';
```
Comment (pengx17) [inline packages/hooks/src/use-block-suite-page-references.ts:5]: I just realized that the way we wrote in `useBlockSuiteWorkspacePage` using weakmap can be replaced by `atomFamily` utility

Comment (himself65) [reply packages/hooks/src/use-block-suite-page-references.ts:5]: yeap

Comment (himself65) [reply packages/hooks/src/use-block-suite-page-references.ts:5]: I want to make it simple so write it manually

**Review by Himself65 — COMMENTED** [review]
Could please add coverage for this line?

https://app.codecov.io/gh/toeverything/AFFiNE/pull/2422/blob/packages/component/src/components/app-sidebar/menu-item/index.tsx#L48

**Review by Himself65 — APPROVED** [review]

**Pull request merged**